### PR TITLE
WIP: #32

### PR DESCRIPTION
**Pull request by SaketJNU**
For Issue No. https://github.com/fabric8-analytics/fabric8-analytics-server/issues/32

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -49,3 +49,4 @@ RUN sh /tmp/update_selinon.sh
 RUN cd /tmp/install_deps/ && /tmp/install_deps/apply_patches.sh
 
 RUN yum install -y npm && npm install -g semver-ranger
+RUN pip3 uninstall -y protobuf && pip3 install packaging appdirs && pip3 install --upgrade --no-binary :all: protobuf==3.2.0
```
Comment (jpopelka) [inline Dockerfile]: Python requirements are specified in [requirements.txt](https://github.com/fabric8-analytics/fabric8-analytics-server/blob/master/requirements.txt), which is generated from [requirements.in](https://github.com/fabric8-analytics/fabric8-analytics-server/blob/master/requirements.in) with `pip-compile --output-file requirements.txt requirements.in`.

File: bayesian/sentiment_analysis.py [hunk 2]
```diff
@@ -0,0 +1,141 @@
...
+        for row in query_data.rows:
+            aggregated_text = aggregated_text + row[0] + ' '
+
+        return  aggregated_text
+
+    def get_latest_comment_time(self, query_data):
+        latest_comment_time = ''
+        for row in query_data.rows:
+            latest_comment_time = row[1]
+        return latest_comment_time
```
Comment (jpopelka) [inline bayesian/sentiment_analysis.py]: You don't need to go through whole list to get the last item, see
https://stackoverflow.com/questions/930397/getting-the-last-element-of-a-list-in-python

so something like the following would be much faster
`return query_data.rows[-1][1] if query_data.rows else ''`

File: requirements.txt [hunk 3]
```diff
@@ -35,3 +35,12 @@ sqlalchemy==1.1.3         # via flask-sqlalchemy
...
 wtforms==2.1              # via flask-wtf
+
+gunicorn
+boto3
+boto
+gremlinpython
+google-cloud-natural-language
+google-cloud-bigquery
+protobuf
+pyyaml
```
Comment (jpopelka) [inline requirements.txt]: Use `pip-compile --output-file requirements.txt requirements.in`

File: requirements.in [hunk 4]
```diff
@@ -16,3 +16,11 @@ pyjwt
 requests-futures
 # we require cucoslib, but it is installed manually as it is not available in PyPI
 #cucoslib
+gunicorn
+boto3
+boto
```
Comment (jpopelka) [inline requirements.in]: I don't see where you use boto(3).

**Comment by samuzzal-choudhury** [comment]
@SaketJNU does this PR still hold good? If not, can you please close this?

**Comment by SaketJNU** [comment]
This work has been included by some other PR in repos., So closing this to avoid duplication 
Thanks.

**Pull request closed without merging**